packages/maskbook/src/plugins/Trader/UI/trader/Trader.tsx

```diff
@@ -71,39 +66,16 @@ export function Trader(props: TraderProps) {
     } = useTradeStateComputed(provider)
     const { inputToken, outputToken } = tradeStore
 
-    const [inputTokenAddress, setInputTokenAddress] = useState(ETH_ADDRESS)
-    const [outputTokenAddress, setOutputTokenAddress] = useState(address === ETH_ADDRESS ? '' : address)
-
-    const isEtherInput = inputTokenAddress === ETH_ADDRESS
-    const isEtherOutput = outputTokenAddress === ETH_ADDRESS
-
-    const asyncInputTokenDetailed = useTokenDetailed(
-        isEtherInput ? EthereumTokenType.Ether : EthereumTokenType.ERC20,
-        isEtherInput ? ETH_ADDRESS : inputTokenAddress,
-        {
-            name,
-            symbol,
-        },
-    )
-    const asyncOutputTokenDetailed = useTokenDetailed(
-        isEtherOutput ? EthereumTokenType.Ether : EthereumTokenType.ERC20,
-        isEtherOutput ? ETH_ADDRESS : outputTokenAddress,
-        {
-            name,
-            symbol,
-        },
-    )
-
     useEffect(() => {
         dispatchTradeStore({
             type: TradeActionType.UPDATE_INPUT_TOKEN,
-            token: asyncInputTokenDetailed.value,
+            token: fromToken,
         })
         dispatchTradeStore({
             type: TradeActionType.UPDATE_OUTPUT_TOKEN,
-            token: asyncOutputTokenDetailed.value,
+            token: toToken,
```

Comment:
The `decimals` should read from the chain.

Reply:
Read decimals from the result of  `useTokenDetailed()` rather than from `trending.coin`, right?

Reply:
DONE